packages/babel-plugin-transform-es2015-modules-umd/src/index.js

```diff
@@ -65,23 +74,74 @@ export default function ({ types: t }) {
             } else if (arg.value === "exports") {
               return t.memberExpression(t.identifier("mod"), t.identifier("exports"));
             } else {
-              let requireName = basename(arg.value, extname(arg.value));
-              let globalName = browserGlobals[requireName] || requireName;
+              let memberExpression;
 
-              return t.memberExpression(t.identifier("global"), t.identifier(
-                t.toIdentifier(globalName)
-              ));
+              if (state.opts.exactGlobals) {
+                let globalRef = browserGlobals[arg.value];
+                if (globalRef) {
+                  if (globalRef.indexOf(".") > -1) {
+                    memberExpression = globalRef.split(".").reduce(
+                      (accum, curr) => t.memberExpression(accum, t.identifier(curr)), t.identifier("global")
+                    );
+                  } else {
+                    memberExpression = t.memberExpression(t.identifier("global"), t.identifier(globalRef));
+                  }
+                } else {
+                  memberExpression = t.memberExpression(
+                    t.identifier("global"), t.identifier(t.toIdentifier(arg.value))
+                  );
+                }
+              } else {
+                let requireName = basename(arg.value, extname(arg.value));
+                let globalName = browserGlobals[requireName] || requireName;
+                memberExpression = t.memberExpression(
+                  t.identifier("global"), t.identifier(t.toIdentifier(globalName))
+                );
+              }
+
+              return memberExpression;
             }
           });
 
-          let globalArg = t.identifier(t.toIdentifier(moduleName ? moduleName.value : this.file.opts.basename));
+          let moduleNameOrBasename = moduleName ? moduleName.value : this.file.opts.basename;
+          let globalToAssign = t.memberExpression(
+            t.identifier("global"), t.identifier(t.toIdentifier(moduleNameOrBasename))
+          );
+          let prerequisiteAssignments = null;
+
+          if (state.opts.exactGlobals) {
+            let globalName = browserGlobals[moduleNameOrBasename];
+
+            if (globalName) {
+              if (globalName.indexOf(".") > -1) {
+                prerequisiteAssignments = [];
+
+                let members = globalName.split(".");
+                let namespacedProperties = members.slice(1).reduce((accum, curr, index) => {
+                  let prerequisiteAssignment = buildPrerequisiteAssignment({ GLOBAL_REFERENCE: accum[index] });
+                  prerequisiteAssignments.push(prerequisiteAssignment);
+                  accum.push(t.memberExpression(accum[index], t.identifier(curr)));
+                  return accum;
```

Comment:
You only ever use the most recent item in `accum` if I'm reading right, so could this be

```
globalToAssign = members.slice(1).reduce((accum, curr) => {
    prerequisiteAssignments.push(buildPrerequisiteAssignment({ GLOBAL_REFERENCE: accum }));
    return t.memberExpression(accum, t.identifier(curr));
}, t.memberExpression(t.identifier("global"), t.identifier(members[0])));
```